File: SynergyGig/src/main/java/entities/Course.java
package entities;

public class Course {
    private int id;
    private String title;
    private String description;
    private int instructorId;
    private int skillId; // Added for Skill Integration

    public Course() {
    }

    public Course(int id, String title, String description, int instructorId, int skillId) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.instructorId = instructorId;
        this.skillId = skillId;
    }

    public Course(String title, String description, int instructorId, int skillId) {
        this.title = title;
        this.description = description;
        this.instructorId = instructorId;
        this.skillId = skillId;
    }

    // Keep constructors for backward compatibility if needed, or update callers
    public Course(String title, String description, int instructorId) {
        this.title = title;
        this.description = description;
        this.instructorId = instructorId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getInstructorId() {
        return instructorId;
    }

    public void setInstructorId(int instructorId) {
        this.instructorId = instructorId;
    }

    public int getSkillId() {
        return skillId;
    }

    public void setSkillId(int skillId) {
        this.skillId = skillId;
    }

    @Override
    public String toString() {
        return "Course{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", instructorId=" + instructorId +
                ", skillId=" + skillId +
                '}';
    }
}

File: src/Entity/Course.php
<?php

namespace App\Entity;

use App\Repository\CourseRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: CourseRepository::class)]
#[ORM\Table(name: "courses")]
class Course
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $title = null;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $description = null;

    #[ORM\Column(nullable: true)]
    private ?int $instructor_id = null;

    #[ORM\Column(nullable: true)]
    private ?int $skill_id = null;

    #[ORM\Column(length: 50, options: ["default" => "Beginner"])]
    private ?string $skill_level = "Beginner";

    #[ORM\OneToMany(mappedBy: 'course', targetEntity: Quiz::class, orphanRemoval: true)]
    private Collection $quizzes;

    public function __construct()
    {
        $this->quizzes = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle(string $title): self
    {
        $this->title = $title;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getInstructorId(): ?int
    {
        return $this->instructor_id;
    }

    public function setInstructorId(?int $instructor_id): self
    {
        $this->instructor_id = $instructor_id;
        return $this;
    }

    public function getSkillId(): ?int
    {
        return $this->skill_id;
    }

    public function setSkillId(?int $skill_id): self
    {
        $this->skill_id = $skill_id;
        return $this;
    }

    public function getSkillLevel(): ?string
    {
        return $this->skill_level;
    }

    public function setSkillLevel(string $skill_level): self
    {
        $this->skill_level = $skill_level;
        return $this;
    }

    /**
     * @return Collection<int, Quiz>
     */
    public function getQuizzes(): Collection
    {
        return $this->quizzes;
    }

    public function addQuiz(Quiz $quiz): self
    {
        if (!$this->quizzes->contains($quiz)) {
            $this->quizzes->add($quiz);
            $quiz->setCourse($this);
        }

        return $this;
    }

    public function removeQuiz(Quiz $quiz): self
    {
        if ($this->quizzes->removeElement($quiz)) {
            // set the owning side to null (unless already changed)
            if ($quiz->getCourse() === $this) {
                $quiz->setCourse(null);
            }
        }

        return $this;
    }
}
